torch/csrc/jit/tensorexpr/llvm_codegen.cpp

```diff
@@ -720,7 +720,8 @@ void LLVMCodeGenImpl::visit(const Max* v) {
   auto rhs = this->value_;
 
   if (v->dtype().is_integral()) {
-    auto icmp = irb_.CreateICmpSGT(lhs, rhs);
+    auto icmp = v->dtype().is_signed() ? irb_.CreateICmpSGT(lhs, rhs)
```

Comment:
You could use `CreateICmp` and `llvm_comparison_predicate` here and avoid the ternary - I had to do a similar fix for comparisons.